packages/near-api-js/src/account.ts

```diff
@@ -293,13 +294,18 @@ export class Account {
         }
 
         try {
-            const accessKey = await this.connection.provider.query<AccessKeyView>({
+            const rawAccessKey = await this.connection.provider.query<AccessKeyViewRaw>({
```

Comment:
Would it be better to place this coercion logic within `JsonRpcProvider`? `nonce` is only really used in this file, but if we were to use these same `query` `methods` else where, we'd most likely have to duplicate this logic. Similarly, someone adding a new `query` type in `JsonRpcProvider` which contains `nonce` may not know that coercion is necessary because it happens within this file.

We'd have to inspect the incoming `request_type` which doesn't seem ideal, but I think the centralization of it would be beneficial.

Thoughts?

Reply:
First of all, changes on this PR is based on the mindset of `nonce` should be returned as a `string` NOT a `number`. (It is currently returning as a number) The problem with returning with number is that, javascript `number` type is less than what max `u64` can be as an integer. So at some point, it will be essential for RPC to make this change for the sake of precision on `nonce`. 

So, changes on this PR allows to support both `number` and `string` type of nonce where it will become a future proof regardless when RPC is making the `nonce` type change. So assuming that `nonce` is returned as `string` that is larger than js `number` integer, in order to increment, we can't just use `++`. So it should still need to be wrapped with `new BN`, wherefore this logic is actually unavoidable.

Reply:
What I am suggesting here is we move this logic in to `JsonRpcProvider` so that it is encapsulated and will be reused if these queries are ever used elsewhere.

But on second thought it may be better to leave this as is, and deal with that case if/when we need.